### PR TITLE
fix(@cubejs-client/core): improve types

### LINKED ISSUE #350: [`@cubejs-client/core`] Types for `ResultSet` are too strict
**Describe the bug**
The `pivotConfig` parameter should be optional

**To Reproduce**
1. Use Typescript
2. Try compiling `resultSet.tablePivot()`
3. See error

**Expected behavior**
The compiler should not throw an error saying `pivotConfig` is a required object.

**Screenshots**
Screen Shot 2020-01-15 at 22 21 57

**Version:**
0.13.9

### PR DESCRIPTION
**Pull request by hassankhan**
**Check List**
- [X] Tests has been run in packages where changes made if available
- [X] Linter has been run for changed code
- [X] Tests for the changes have been added if not covered yet
- [X] Docs have been added / updated if required

**Issue Reference this PR resolves**

Fixes #350

**Description of Changes Made (if issue reference is not provided)**

Loosened types as appropriate.

### CODE REVIEW

File: packages/cubejs-client-core/dist/index.d.ts [hunk 1]
```diff
@@ -82,24 +97,36 @@ export type Query = {
...
 
-  load(query: Query, options, callback): void;
-  load(query: Query, options): Promise<ResultSet>;
+  load(query: Query, options?: LoadMethodOptions, callback?: LoadMethodCallback): void;
+  load(query: Query, options?: LoadMethodOptions): Promise<ResultSet>;
 
-  sql(query: Query, options, callback): void;
-  sql(query: Query, options): Promise<ResultSet>;
+  sql(query: Query, options?: LoadMethodOptions, callback?: LoadMethodCallback): void;
+  sql(query: Query, options?: LoadMethodOptions): Promise<ResultSet>;
```
Comment (paveltiunov) [inline packages/cubejs-client-core/dist/index.d.ts]: `ResultSet` should be replaced here with something

File: packages/cubejs-client-core/dist/index.d.ts [hunk 2]
```diff
@@ -82,24 +97,36 @@ export type Query = {
...
 
-  meta(options, callback): void;
-  meta(options): Promise<ResultSet>;
+  meta(options?: LoadMethodOptions, callback?: LoadMethodCallback): void;
+  meta(options?: LoadMethodOptions): Promise<ResultSet>;
```
Comment (paveltiunov) [inline packages/cubejs-client-core/dist/index.d.ts]: `ResultSet` should be replaced here with something

File: packages/cubejs-client-core/dist/index.d.ts (hunk 1 above)
Comment (hassankhan) [reply packages/cubejs-client-core/dist/index.d.ts]: Hi @paveltiunov, not quite sure what you mean by that? What should `ResultSet` be replaced by?

Comment (paveltiunov) [reply packages/cubejs-client-core/dist/index.d.ts]: @hassankhan `sql()` method returns `SqlQuery` instances of class and not `ResultSet` with query results. Same for `meta()`: it returns `Meta` instances of class.

Comment (hassankhan) [reply packages/cubejs-client-core/dist/index.d.ts]: @paveltiunov Added types for `SqlQuery`

File: packages/cubejs-client-core/dist/index.d.ts (hunk 2 above)
Comment (hassankhan) [reply packages/cubejs-client-core/dist/index.d.ts]: @paveltiunov Added types for `Meta`

**Comment by paveltiunov** [comment]
@hassankhan Can we hold on with publishing `MetaApiResponse` and `SqlApiResponse`? Those aren't public API yet.

**Comment by hassankhan** [comment]
I can remove them, but I somewhat disagree with it being a private API. Personally I'd prefer to leave them in for the time being, and update as appropriate.

Happy to remove them if you'd still prefer me to, though 👍

**Comment by paveltiunov** [comment]
@hassankhan I mean those API responses aren't intended to be accessed directly. Only through `SqlQuery` and `Meta` classes.

**Pull request merged**